packages/dds-components/src/components/date-inputs/DatePicker/DateField/DateField.tsx

```diff
@@ -53,6 +59,27 @@ export function DateField({
   const { labelProps, fieldProps } = useDateField(props, state, internalRef);
 
   const disabled = props.isDisabled || !!fieldProps['aria-disabled'];
+  const showClearButton = clearable && !disabled && !props.isReadOnly;
+
+  const [hasValue, setHasValue] = useState(!!state.value);
+  const clearButtonSize = componentSize === 'medium' ? 'medium' : 'small';
+
+  useEffect(() => {
+    setHasValue(!!state.value);
+  }, [state.value]);
+
+  useEffect(() => {
+    const hasPartialValue = state.segments.some(
+      segment =>
+        segment.type !== 'literal' && segment.placeholder !== segment.text,
+    );
+    setHasValue(hasPartialValue);
+  }, [state.segments]);
+
+  const clearDate = () => {
+    state.setValue(null);
+    setHasValue(false);
+  };
```

Comment:
Trenger kanskje ikke state for `hasValue`


```suggestion
  const hasPartialValue = state.segments.some(segment => segment.type !== 'literal' && segment.placeholder !== segment.text)
  const hasValue = !!state.value || hasPartialValue
  const clearButtonSize = componentSize === 'medium' ? 'medium' : 'small';

  const clearDate = () => {
    state.setValue(null);
  };
```

Reply:
fair enough!